test/box-luatest/gh_8051_set_box_cfg_thru_env_test.lua

```diff
@@ -0,0 +1,93 @@
+local server = require('luatest.server')
+local popen = require('popen')
+local clock = require('clock')
+
+local t = require('luatest')
+local g = t.group()
+
+g.after_all = function()
+    if g.server ~= nil then
+       g.server:stop()
+   end
+end
+
+local TARANTOOL_PATH = arg[-1]
+
+local function popen_run(command)
+    local cmd = TARANTOOL_PATH .. ' -i 2>&1'
+    local ph = popen.new({cmd}, {
+        shell = true,
+        setsid = true,
+        group_signal = true,
+        stdout = popen.opts.PIPE,
+        stderr = popen.opts.PIPE,
+        stdin = popen.opts.PIPE,
+    })
+    t.assert(ph, 'process is not up')
+
+    ph:write(command)
+
+    local output = '';
+    local time_quota = 2.0
+    local start_time = clock.monotonic();
+    while output:find('message') == nil
+        and clock.monotonic() - start_time < time_quota do
+        local data = ph:read({timeout = 1.0})
+        if data ~= nil then
+            output = output .. data
+        end
```

Comment:
EOF is not handled?

Reply:
What do you mean?

Reply:
> Return:
>
> <...> empty string if EOF

https://www.tarantool.io/en/doc/latest/reference/reference_lua/popen/#popen-read

This case is not handled here AFAIS, so we'll spin in the loop till the time quota will be reached. However there is no sense to do so when we got EOF.

Reply:
Done:
```
      while clock.monotonic() - start_time < time_quota do
          local chunk = ph:read({timeout = 1.0})
          if chunk == '' or chunk == nil then                                                                       
              -- EOF or error 
              break                                                                                                 
          end                                                                                                       
          output = output .. chunk                                                                                  
      end                                                                                                           
```

Reply:
I would raise an error in case of a `:read()` error, but just return may work for this given test. Closing the discussion.